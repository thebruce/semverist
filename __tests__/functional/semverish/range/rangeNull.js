const rangeFactory = require('../../../../lib/semverish/range');

const tmpConfig = {
  semveristBehaviors: {
    inheritence: null,
    lazySemverist: {
      attribute: true,
      preReleaseForwards: false,
    },
    default: true,
    defaultName: 'default',
    groups: true,
    mergeStrategy: 'lastIn',
    preReleasePattern: /\d-[a-zA-Z]*/g,
  },
  groups: {},
  prereleaseOrdering: {},
};
describe('Range null tests.', () => {
  test('setSemveristRangeNull', () => {
    expect.assertions(1);
    return rangeFactory('semverist', 'range')
      .then(RangeClass => {
        const range = new RangeClass();
        range.init(tmpConfig);
        range.setOptions();
        range.setLowerBounds('1.0.0');
        range.setSemveristElement('entity');
        range.setSemveristElementType('attribute');
        range.setSemverish('1');
        range.setSemverishArray('1');
        range.setSemver('1.0.0');
        range.setExceptions();
        range.addException('1.1');
        range.setRange();
        range.setTerminalBounds('<2.0.0');
        range.setSemveristRange();
        return range.getSemveristRange();
      })
      .then(obj => expect(obj).toMatchSnapshot());
  });
});
