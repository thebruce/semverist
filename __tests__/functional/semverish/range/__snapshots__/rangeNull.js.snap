// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range null tests. setSemveristRangeNull 1`] = `
Object {
  "adjustedExceptions": Array [],
  "exceptionRange": null,
  "exceptions": Array [],
  "lowerBounds": "1.0.0",
  "range": "1.0.0",
  "semverishValue": "1",
  "semveristElement": "entity",
  "semveristElementType": "attribute",
  "terminalBounds": "<2.0.0",
}
`;
